tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs

```diff
@@ -139,6 +159,49 @@ public async Task JsonWebTokenWithX509PublicCertSendCertificateOnBehalfOfTestAsy
             }
         }
 
+        [TestMethod]
+        [Description("Test for client assertion with X509 public certificate using sendCertificate")]
+        public async Task JsonWebTokenWithX509PublicCertSendCertificateSilentTestAsync()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                SetupMocks(harness.HttpManager, "https://login.microsoftonline.com/my-utid/");
+                var certificate = new X509Certificate2(
+                    ResourceHelper.GetTestResourceRelativePath("valid_cert.pfx"),
+                    MsalTestConstants.DefaultPassword);
+
+                var app = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                              .WithAuthority(
+                                                                  new System.Uri("https://login.microsoftonline.com/my-utid"),
+                                                                  true).WithRedirectUri(MsalTestConstants.RedirectUri)
+                                                              .WithHttpManager(harness.HttpManager)
+                                                              .WithCertificate(certificate).BuildConcrete();
+
+
+
+                _tokenCacheHelper.PopulateCache(app.UserTokenCacheInternal.Accessor);
```

Comment:
Would be better to mock an actual call instead of populating the token cache artificially.

Reply:
Not sure I understand the reason why? The functionality that is being testing is whether or not the x5c claim is present in the header of the acquire token silent request. How the user token made its way to the cache is irrelevant. Also, there are other acquire token silent tests that do the same thing.

Reply:
Not blocking the fix on this, but the arguments would be: 

- artificially populating the cache makes it difficult to the test reader to understand what's in the cache
- cache population logic is quite likely to become different than how MSAL populates the cache (e.g. pretty sure this won't add AppMetadata which is a FOCI concept). 
- overall test logic is cleaner because it's closer to what the user would do, i.e. fetch a token, fetch smth silent. The user does not "populate the cache".